torch/nn/functional.py

```diff
@@ -2439,6 +2439,7 @@ def layer_norm(
     weight: Optional[Tensor] = None,
     bias: Optional[Tensor] = None,
     eps: float = 1e-5,
+    out: Optional[Tensor] = None,
```

Comment:
I'm not sure we want that. I don't think any functional function has out arguments

Reply:
maybe they should? from my experience, there's no big difference between core and functional functions besides historical reasons or just reducing surface of core namespace.

doing out-variant is a useful intermediate step for implementing an inplace autograd function: https://github.com/pytorch/pytorch/issues/26288

Reply:
cc @jbschlosser for the more general question here?

Reply:
IIRC there is one function in `torch.nn.functional` that takes an `out` param: `normalize()`.

Imo it's not great that the functional wrappers in general restrict users from accessing the out variants. I think ideally either the functional wrappers wouldn't exist, or they would consistently allow for `out` to be specified.

If there's no JIT / other reason preventing us from supporting `out` here, I think we should. Even better - is it possible to remove the `torch.nn.functional.layer_norm` definition and use `_add_docstr()` instead?

Reply:
Agree with @jbschlosser, looks like this is a no-op wrapper, so we shouldn't have it and should have `_add_docstr()`

Reply:
sounds good!

Reply:
Update: turns out there is a purpose for this wrapper wrt FX. From @jamesr66a:

> yeah the problem here is that tracing functionals is more permissive than tracing ops. When tracing functionals, we can accept a proxy argument for any parameter. when tracing ops, the PythonArgParser will only do _torch_function_ dispatch if proxys appear for parameters that are of Tensor or TensorList type. So by removing the functional and replacing it with an op, this is causing a traceability regression.
I've complained about the issue with tracing ops, but the core team said they do it for performance reasons. Maybe you could try bugging them again about this?

Problem is with the `normalized_size` arg since it's not of type `Tensor` or `TensorList`. If we want to keep FX traceability, I think we have to keep the wrapper here.